e2e/brickmux_test.go

```diff
@@ -276,5 +276,176 @@ func TestBrickMux(t *testing.T) {
 		r.Nil(client.VolumeStop(volname1 + strconv.Itoa(i)))
 		r.Nil(client.VolumeDelete(volname1 + strconv.Itoa(i)))
 	}
+
+	// Turn on brick mux cluster option
+	optReq = api.ClusterOptionReq{
+		Options: map[string]string{"cluster.max-bricks-per-process": "2"},
+	}
+	err = client.ClusterOptionSet(optReq)
+	r.Nil(err)
+
+	for i := 37; i <= 54; i++ {
+		brickPath := testTempDir(t, "brick")
+		brickPaths = append(brickPaths, brickPath)
+	}
+
+	/* Test  for Testing max-bricks-per-process constraint while
+	multiplexing*/
+
+	// Create 5 volumes and start all 5
+	// making all brick multiplexed with the constraint that
+	// max-bricks-per-process is 2
+
+	index = 37
+	for i := 1; i <= 5; i++ {
+		createReq := api.VolCreateReq{
+			Name: volname1 + strconv.Itoa(i),
+			Subvols: []api.SubvolReq{
+				{
+					Type: "distribute",
+					Bricks: []api.BrickReq{
+						{PeerID: tc.gds[0].PeerID(), Path: brickPaths[index]},
+						{PeerID: tc.gds[0].PeerID(), Path: brickPaths[index+1]},
+					},
+				},
+			},
+			Force: true,
+		}
+		_, err = client.VolumeCreate(createReq)
+		r.Nil(err)
+
+		// start the volume
+		err = client.VolumeStart(volname1+strconv.Itoa(i), false)
+		r.Nil(err)
+
+		index = index + 2
+	}
+
+	// pidMap and portMap just to mantain count of every pid and port of
+	// bricks from all 5 volumes.
+	pidMap := make(map[int]int)
+	portMap := make(map[int]int)
+	for i := 1; i <= 5; i++ {
+		bstatus, err = client.BricksStatus(volname1 + strconv.Itoa(i))
+		r.Nil(err)
+		for _, b := range bstatus {
+			if _, ok := pidMap[b.Pid]; ok {
+				pidMap[b.Pid]++
+			} else {
+				pidMap[b.Pid] = 1
+			}
+
+			if _, ok := portMap[b.Port]; ok {
+				portMap[b.Port]++
+			} else {
+				portMap[b.Port] = 1
+			}
+
+		}
+	}
+
+	// Check if all pid's and ports's have count = 2 as mentioned in
+	// max-bricks-per-process
+	for _, v := range pidMap {
+		r.Equal(v, 2)
+	}
+
+	for _, v := range portMap {
+		r.Equal(v, 2)
+	}
+
+	for i := 1; i <= 5; i++ {
+		r.Nil(client.VolumeStop(volname1 + strconv.Itoa(i)))
+		r.Nil(client.VolumeDelete(volname1 + strconv.Itoa(i)))
+	}
+
+	// Create two volumes with different options, so that bricks from these
+	// two volumes are multiplexed into bricks from their own volume. Also,
+	// check if among three bricks of a volume 2 bricks have same pid and
+	// port while 1 brick has a different pid and port, since num  of bricks
+	// are 3 and max-bricks-per-process is set as 2.
+	createReq = api.VolCreateReq{
+		Name: volname1,
+		Subvols: []api.SubvolReq{
+			{
+				ReplicaCount: 3,
+				Type:         "replicate",
```

Comment:
create multiple replicate volumes and do the check for max-bricks-per-process.

Reply:
Done